### ISSUE TITLE
C2x standard not recognized.

### ISSUE THREAD

**Issue by michaelbrockus** [mesonbuild/meson#7970]
**Describe the bug**
I was hoping to try out the new C2x standard when I stumbled upon it not being recognized by Meson.

**To Reproduce**

```meson
project(‘c21 experimental’, ‘c’, default_options : [‘c_std=c2x’])

executable(‘prog’, ‘main.c’)
```

**Expected behavior**
I as the user expect to use the `c_std` option with the value of `c2x` with it being recognized so I don’t have to manually add the argument myself through add arguments command.

**system parameters**
* This a plain native build (for the same computer)?
* Operating systems MacOS Catalina, and Windows 10
* what `python 3.9.0`
* what `meson 0.56.0`
* what `ninja 1.10.0.post2`

**Comment by michaelbrockus** [comment]
As I understand it the C2x standard was implemented in `gcc` 9.0, and `clang’ 9.0.

**Comment by michaelbrockus** [comment]
Never mind I’m mistaken this issue with 0.55.3.
